packages/types/src/client.ts

```diff
@@ -2,6 +2,16 @@ import { Command } from "./command";
 import { MiddlewareStack } from "./middleware";
 import { MetadataBearer } from "./response";
 
+export type Exact<Type1, Type2> = [Type1] extends [Type2] ? ([Type2] extends [Type1] ? true : false) : false;
```

Comment:
place this somewhere generic?

Reply:
In util.ts?

Reply:
Moved in [`8d0ebaf` (#858)](https://github.com/awslabs/smithy-typescript/pull/858/commits/8d0ebaf7a81eabe9fca2a8d282416625ec73f85a)